src/pages/kandidat/Kandidatdetaljer.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
+import NavFrontendSpinner from 'nav-frontend-spinner';
+
+import { AppRoute } from '../../utils/paths';
+import { erGyldigFnr, formaterFnr } from '../før-du-begynner/fnr-input/fnrUtils';
+import { hentKandidat, hentSkrivetilgang } from '../../api/api';
+import bemHelper from '../../utils/bemHelper';
+import EndreKandidat from './endre-kandidat/EndreKandidat';
+import EndreKandidatKnapp from './endre-kandidat/EndreKandidatKnapp';
+import Kandidat from '../../types/Kandidat';
+import Placeholderboks from '../../components/placeholderboks/placeholderboks';
+import RouteBanner from '../../components/route-banner/RouteBanner';
+import SeKandidat from './se-kandidat/SeKandidat';
+import Tilbake from '../../components/tilbake/Tilbake';
+import './kandidatdetaljer.less';
+
+const cls = bemHelper('kandidatdetaljer');
+
+const PATHNAME_FNR = 1;
+
+interface OwnProps {
+    endre: boolean;
+}
+
+type Props = OwnProps & RouteComponentProps;
+
+const Kandidatdetaljer = (props: Props) => {
+    const { location, endre } = props;
+
+    const [kandidat, setKandidat] = useState<Kandidat | undefined>(undefined);
+    const [feilmelding, setFeilmelding] = useState<string | undefined>(undefined);
+    const [harSkrivetilgang, settSkrivetilgang] = useState<boolean>(false);
+    const [] = useState<boolean>(false);
+
+    const fnr = location.pathname.split('/')[PATHNAME_FNR];
+
+    const hentKandidatFraUrl = async () => {
+        const gyldigFnr = erGyldigFnr(fnr);
+
+        if (!gyldigFnr) {
+            setFeilmelding('Ugyldig fødselsnummer');
+            return;
+        }
+
+        try {
+            const kandidat = await hentKandidat(fnr);
+            setKandidat(kandidat);
+        } catch (error) {
+            setFeilmelding(
+                'Du har enten ikke tilgang til denne kandidaten eller så finnes ikke kandidaten i systemet'
+            );
+        }
+    };
+
+    const sjekkSkrivetilgang = async () => {
+        try {
+            const harTilgang = await hentSkrivetilgang(fnr);
+            settSkrivetilgang(harTilgang);
+        } catch (error) {
+            settSkrivetilgang(false);
+        }
+    };
+
+    useEffect(() => {
+        hentKandidatFraUrl();
+        sjekkSkrivetilgang();
+    }, [location.pathname]);
+
+    const getMainContent = () => {
```

Comment:
Jeg liker at dette er trukket ut :+1:

Reply:
Kunne brukt et norsk navn for denne funksjonen.

Trenger man ha med `get` (eller hent på norsk) i navnet på slike funksjoner? Når jeg leser `getContent()` i renderfunksjoner synes jeg ofte det er forvirrende da man får inntrykk av at man henter noe data for visning, når man egentlig bare har dratt det ut i en funksjon for at ikke renderfunksjonen blir for lang...

Gir det mening?